src/crawlers/browser_crawler.js

```diff
@@ -360,8 +360,43 @@ export default class BrowserCrawler extends BasicCrawler {
      * @internal
      */
     async _handleRequestFunction(crawlingContext) {
-        const { id } = crawlingContext;
-        const page = await this.browserPool.newPage({ id });
+        const newPageOptions = {
+            id: crawlingContext.id,
+        };
+
+        if (this.proxyConfiguration && this.launchContext.useIncognitoPages) {
+            const pageOptions = {};
+
+            const { session } = crawlingContext;
```

Comment:
I noticed that a new session is generated even if `useIncognitoPages` is set to `false`. Is this expected?

Reply:
Yeah, it's not ideal. `BasicCrawler` is what generates the sessions and it has no notion of `useIncognitoPages`. Technically, it is the correct process to always get a new session for each request. But in case of a browser with persistent context, we can't use those sessions because the proxy is the same for all the requests. I'm happy to change it if you have a better proposal.

Reply:
I think https://sdk.apify.com/docs/api/request could accept a `sessionId`? Or as mentioned on Slack, have a function that can export all session data from that page. This way the session can be reused when the actor needs to migrate.

Reply:
You almost never want to retry the same request with the same session, unless it's some specific request like a login. So tying those two together does not make sense to me semantically. A request is a job to do, a session is a user that we impersonate to do the job. Sessions also die quite fast and need to be replaced with new sessions.

The exporting of data is a separate problem. It would be nice to have for web automation scenarios, but for scraping, you mostly want to look as a lot of different users.

Reply:
So the purpose of incognito pages (in contrast to persistent contexts) is to only use proxy per page?